src/libs/actions/IOU.ts

```diff
@@ -6842,7 +6842,14 @@ function approveMoneyRequest(expenseReport: OnyxEntry<OnyxTypes.Report>, full?:
         total = expenseReport?.unheldTotal;
     }
     const optimisticApprovedReportAction = ReportUtils.buildOptimisticApprovedReportAction(total, expenseReport?.currency ?? '', expenseReport?.reportID ?? '-1');
-    const optimisticNextStep = NextStepUtils.buildNextStep(expenseReport, CONST.REPORT.STATUS_NUM.APPROVED);
+
+    const approvalChain = ReportUtils.getApprovalChain(PolicyUtils.getPolicy(expenseReport?.policyID), expenseReport?.ownerAccountID ?? -1, expenseReport?.total ?? 0);
+    const isLastApprover = approvalChain[approvalChain.length - 1] === currentUserEmail ?? true;
+
+    const predictedNextStatus = isLastApprover ? CONST.REPORT.STATUS_NUM.APPROVED : CONST.REPORT.STATUS_NUM.SUBMITTED;
+    const predictedNextState = isLastApprover ? CONST.REPORT.STATE_NUM.APPROVED : CONST.REPORT.STATE_NUM.SUBMITTED;
```

Comment:
Ok so I think we need to test that this all still works on policies that aren't "control w/ advanced approval".

I have a feeling that "collect, on submit & approve" policies won't have `isLastApprover` ever be `true`, right? Since `ReportUtils.getApprovalChain` returns `[]` if `!PolicyUtils.isControlOnAdvancedApprovalMode(policy)` right?

Let's at least make sure this is tested manually by y'all & Applause

Reply:
> Since ReportUtils.getApprovalChain returns [] if !PolicyUtils.isControlOnAdvancedApprovalMode(policy) right?

In that case `approvalChain[approvalChain.length - 1] ` would be undefined. And `isLastApprover` will default to `true`.

Reply:
Let me make the code a bit more verbose so it's easier to read

Reply:
@Beamanator the code is equivalent to - 

```js
function isLastApprover(approvalChain: string[], currentUserEmail: string): boolean {
    if (approvalChain.length === 0) {
        return true;
    }
    return approvalChain[approvalChain.length - 1] === currentUserEmail;
}

```

Reply:
Tests well